tests/unit/Rules/FactorOfTest.php

```diff
@@ -0,0 +1,133 @@
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+/**
+ * @group  rule
+ * @covers Respect\Validation\Rules\FactorOf
+ * @covers Respect\Validation\Exceptions\FactorOfException
+ */
+class FactorOfTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider providerForValidFactorOf
+     */
+    public function testValidFactorOfShouldReturnTrue($dividend, $input)
+    {
+        $min = new FactorOf($dividend);
+        $this->assertTrue($min->__invoke($input));
+        $this->assertTrue($min->check($input));
+        $this->assertTrue($min->assert($input));
+    }
+
+    /**
+     * @dataProvider providerForInvalidFactorOf
+     */
+    public function testInvalidFactorOfShouldThrowFactorOfException($dividend, $input)
+    {
+        $this->setExpectedException('Respect\Validation\Exceptions\FactorOfException', json_encode($input) . ' must be a factor of ' . json_encode($dividend));
+
+        $min = new FactorOf($dividend);
+        $this->assertFalse($min->__invoke($input));
+        $this->assertFalse($min->assert($input));
+    }
+
+    public function providerForValidFactorOf()
+    {
+        $tests = array(
+            // Run through the first few integers.
+            array(1, 1),
+            array(2, 1),
+            array(2, 2),
+            array(3, 1),
+            array(3, 3),
+            array(4, 1),
+            array(4, 2),
+            array(4, 4),
+            array(5, 1),
+            array(5, 5),
+            array(6, 1),
+            array(6, 2),
+            array(6, 3),
+            array(6, 6),
+        );
+
+        $tests = $this->generateNegativeCombinations($tests);
+
+        $tests = $this->generateStringAndFloatCombinations($tests);
+
+        return $tests;
+    }
+
+    public function providerForInvalidFactorOf()
+    {
+        $tests = array(
+            // Run through the first few integers.
+            array(3, 2),
+            array(4, 3),
+            array(5, 2),
+            array(5, 3),
+            array(5, 4),
+            // Zeros.
+            array(1, 0),
+            array(2, 0),
+        );
+
+        $tests = $this->generateNegativeCombinations($tests);
+
+        $tests = $this->generateStringAndFloatCombinations($tests);
+
+        $extra_tests = array(
+            // Non-integer values.
+            array(1, mt_rand(1, mt_getrandmax() - 1) / mt_getrandmax()),
+            array(1, uniqid('a')),
+        );
+        $tests = array_merge($tests, $extra_tests);
+
+        return $tests;
+    }
+
+    public function providerForInvalidFactorOfDividend()
+    {
+        return array(
+            // Invalid dividend, valid input.
+            array(0, 0),
+            array(0, 1),
+            array(0.5, mt_rand()),
+            array(1.5, mt_rand()),
```

Comment:
1. I disagree, a singular, native function call is hardly difficult to read or understand
2. That's precisely the point, using randomness forces us to write code that is valid for all integers that could be generated and not just the very small subset of integers that I could write by hand

"if it breaks in some point"

yes, there may be hard to reach bugs in places we have not thought of, that only a random test would catch.

"it may be hard to reproduce the exact same error"

Not true, the value that fails should be in the PHPUnit logs, we can see exactly which integer broke and then add that as a special/edge case explicitly in the tests at that time.